nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -90,4 +123,177 @@ private static void invalidInstantiationError(
         errorBuilder.createErrorDescription(
             errorMessage, analysis.buildDescription(tree), state, null));
   }
+
+  @SuppressWarnings("UnusedVariable")
+  public static void checkInstantiationForAssignments(
+      Tree tree, Config config, VisitorState state, NullAway analysis) {
+    Tree lhsTree;
+    Tree rhsTree;
+    if (tree instanceof VariableTree) {
+      VariableTree varTree = (VariableTree) tree;
+      lhsTree = varTree.getType();
+      rhsTree = varTree.getInitializer();
+    } else {
+      AssignmentTree assignmentTree = (AssignmentTree) tree;
+      lhsTree = assignmentTree.getVariable();
+      rhsTree = assignmentTree.getExpression();
+    }
+    if (rhsTree == null) {
+      // Possible for VariableTrees with no initializer
+      return;
+    }
+    if (rhsTree instanceof NewClassTree
+        && ((NewClassTree) rhsTree).getIdentifier() instanceof ParameterizedTypeTree) {
+      ParameterizedTypeTreeNullableArgIndices typeWrapper =
+          new ParameterizedTypeTreeNullableArgIndices();
+      typeWrapper.checkAssignmentTypeMatch(
+          tree,
+          ASTHelpers.getType(lhsTree),
+          (ParameterizedTypeTree) ((NewClassTree) rhsTree).getIdentifier(),
+          config,
+          state,
+          analysis);
+    } else {
+      NormalTypeTreeNullableTypeArgIndices typeWrapper = new NormalTypeTreeNullableTypeArgIndices();
+      typeWrapper.checkAssignmentTypeMatch(
+          tree, ASTHelpers.getType(lhsTree), ASTHelpers.getType(rhsTree), config, state, analysis);
+    }
+  }
+}
+
+class ParameterizedTypeTreeNullableArgIndices
```

Comment:
We should not have multiple top-level classes in the same source file.  These should be `static` nested classes of `GenericsChecks`.  I enabled the Error Prone check that catches this.